airflow/models/dagrun.py

```diff
@@ -729,6 +726,10 @@ def _get_ready_tis(
                     additional_tis.extend(expanded_tis[1:])
                 expansion_happened = True
             if schedulable.state in SCHEDULEABLE_STATES:
+                if schedulable.task.is_mapped:
+                    task = cast("MappedOperator", schedulable.task)
```

Comment:
```suggestion
                task = schedulable.task
                if isinstance(task, MappedOperator):
```

since we are dealing with MappedOperator explicitly, using `isinstance` is more correct here.

Why are we putting this method on MappedOperator though? The logic does not really need much from the operator and the previous way putting the logic on DagRun makes more sense to me.

Reply:
Well, no much reason, it just gave me a clear head to think about what was going on, I felt it was better there after because of the cache on `task.run_time_mapped_ti_count` but I'm returning it back to dagrun.